pkg/cloud/aws/services/userdata/controlplane.go

```diff
@@ -157,14 +213,65 @@ type ContolPlaneJoinInput struct {
 	CACertHash     string
 	CACert         string
 	CAKey          string
+	EtcdCert       string
+	EtcdKey        string
+	FrontProxyCert string
+	FrontProxyKey  string
+	SaCert         string
+	SaKey          string
 	BootstrapToken string
 	ELBAddress     string
-	KubeConfig     string
+}
+
+func (cpi *ControlPlaneInput) isValid() (bool, string) {
```

Comment:
this should return just error and all of these cases need to return the reason wrapped in `errors.New`

Reply:
fixed.